pkg_resources/__init__.py

```diff
@@ -3323,6 +3311,26 @@ def _initialize_master_working_set():
     globals().update(locals())
 
 
+if TYPE_CHECKING:
+    # All of these are set by the @_call_aside methods above
+    __resource_manager = ResourceManager()  # Won't exist at runtime
+    resource_exists = __resource_manager.resource_exists
+    resource_isdir = __resource_manager.resource_isdir
+    resource_filename = __resource_manager.resource_filename
+    resource_stream = __resource_manager.resource_stream
+    resource_string = __resource_manager.resource_string
+    resource_listdir = __resource_manager.resource_listdir
+    set_extraction_path = __resource_manager.set_extraction_path
+    cleanup_resources = __resource_manager.cleanup_resources
+
+    working_set = WorkingSet()
+    require = working_set.require
+    iter_entry_points = working_set.iter_entry_points
+    add_activation_listener = working_set.subscribe
+    run_script = working_set.run_script
+    run_main = run_script
```

Comment:
My question here is:

Instead of "re-declaring" the global variables, wouldn't it make more sense to ditch the `@_call_aside` and just run the majority of the body of `_initialize*` functions directly on the global scope?

@jaraco, do you know if `_call_aside()`/`_initialize_master_working_set()`/`_initialize()` is used in `pkg_resources` just for the sake of organising the code, or is this very small delay on the execution a trick that solves a specific problem?

(I do like the way the code is organised right now, but if changes are needed for type-checking, I would prefer not having double book keeping. Of course we can decide that the type-checking here is not worth the trouble).

Reply:
> I would prefer not having double book keeping

FWIW, you already have to for other checkers (just as a note, these variables were moved from the top of the file, so this PR doesn't introduce double book-keeping).

That being said, I agree it'd be nicer if we didn't have to, but I wouldn't want to revamp `pkg_resources` too much outside of making it a fully typed package to take it out of typeshed.